File: solutions/typescript/matrix/1/matrix.ts
class Matrix {
	readonly rows: Array<Array<number>>
	readonly columns: Array<Array<number>>

	constructor(matrix: string) {
		this.rows = matrix.split("\n").map((row) => row.split(" ").map(Number))
		this.columns = Array.from({ length: this.rows[0].length }, () => [])
		for (const [index, column] of this.columns.entries()) {
			for (const row of this.rows) {
				column.push(row[index])
			}
		}
	}
}

export { Matrix }

File: solutions/scala/matrix/5/src/main/scala/Matrix.scala
import Matrix.toInts

case class Matrix(s: String) {
  private val rows = s.split("\\R").map(toInts).toVector
  private val columns = rows.transpose

  def row(n: Int): Seq[Int] = rows(n)
  def column(n: Int): Seq[Int] = columns(n)
}

object Matrix {
  private def toInts(s: String) = s.split("\\s").map(_.toInt).toVector
}
